### PR TITLE
DOCS-2257 Restrictions by IP

### PR DESCRIPTION
**Pull request by MaximBashurov**

### CODE REVIEW

File: docs/latest/user-guides/ip-lists/overview.md [hunk 1]
```diff
@@ -8,30 +8,94 @@ In the **IP lists** section of Wallarm Console, you can control access to your a
...
+* [GreenSnow](https://blocklist.greensnow.co/greensnow.txt)
+* [www.blocklist.de](https://www.blocklist.de/en/export.html)
+* [NGINX ultimate bad bot blocker](https://github.com/mitchellkrogza/nginx-ultimate-bad-bot-blocker/blob/master/_generator_lists/bad-ip-addresses.list)
+* [IPsum](https://github.com/stamparm/ipsum)
+
+## Limiting by target application
+
+When adding object to a list, by default it will process all requests from the listed IP. But you can limit that by target applications: select one or several applications and only requests from the listed IP to that applications will be processed.
+
+## IP lists over time
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/ip-lists/overview.md:51]: it is called history, please use the same naming as we do in the UI

Comment (AnastasiaTWW) [reply docs/latest/user-guides/ip-lists/overview.md:51]: the things are made up here. reviewing history and changing period for IP to be in the list are different cases

File: docs/latest/user-guides/ip-lists/overview.md [hunk 2]
```diff
@@ -8,30 +8,94 @@ In the **IP lists** section of Wallarm Console, you can control access to your a
 
 ![All IP lists](../../images/user-guides/ip-lists/ip-lists-home-apps.png)
 
-## Algorithm of IP lists processing
+## How allowlist, denylist, and graylist work together
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/ip-lists/overview.md:11]: it should go not before the section about populating lists as it prevents users from using IP lists quicker

File: docs/latest/user-guides/ip-lists/overview.md [hunk 3]
```diff
@@ -8,30 +8,94 @@ In the **IP lists** section of Wallarm Console, you can control access to your a
...
+* Graylist is only considered in `Safe blocking` mode.
 
-1. If Wallarm node is located behind a load balancer or CDN, please make sure to configure your Wallarm node to properly report end-user IP addresses:
+!!! warning "Exceptions"
+    If [`wallarm_acl_access_phase off`](../../admin-en/configure-parameters-en.md#wallarm_acl_access_phase), the Wallarm node does not analyze the denylist in the `off` mode and does not block requests from denylisted IPs in the `Monitoring` mode.
 
-    * [Instructions for NGINX-based Wallarm nodes](../../admin-en/using-proxy-or-balancer-en.md) (including AWS / GCP images and Docker node container)
-    * [Instructions for the filtering nodes deployed as the Wallarm Kubernetes Ingress controller](../../admin-en/configuration-guides/wallarm-ingress-controller/best-practices/report-public-user-ip.md)
-2. Add request sources to IP lists:
+## Listing IPs, subnets, locations, and source types
```
[the remark this answers is not in the document]
Comment (AnastasiaTWW) [reply docs/latest/user-guides/ip-lists/overview.md:28]: in this section, describe that there are two ways to populate IP lists: manual and automatic

File: docs/latest/user-guides/ip-lists/overview.md [hunk 4]
```diff
@@ -8,30 +8,94 @@ In the **IP lists** section of Wallarm Console, you can control access to your a
...
-!!! warning "Using additional traffic filtering facilities"
-    Note that if you use additional facilities (software or hardware) to automatically filter and block traffic, it is recommended that you configure an allowlist with the IP addresses for the [Wallarm Scanner](../../about-wallarm/detecting-vulnerabilities.md#vulnerability-scanner). This will allow Wallarm components to seamlessly scan your resources for vulnerabilities.
+* **IP or subnet** - the supported maximum subnet mask is `/32` for IPv6 addresses and `/12` for IPv4 addresses.
 
-    * [Scanner IP address registered in Wallarm US Cloud](../../admin-en/scanner-address-us-cloud.md)
-    * [Scanner IP address registered in Wallarm EU Cloud](../../admin-en/scanner-address-eu-cloud.md)
+* **Location** (country or region) to add all IP addresses registered in this country or region
+* **Source type** to add all IP addresses that belong to this type, e.g. Tor, AWS, VPN and others.
+
+Note that **Malicious IPs** source type covers IP addresses that are well-known for malicious activity, as mentioned in public sources, and verified by expert analysis. We pull this data from a combination of the following resources:
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/ip-lists/overview.md:37]: it relates to the last item from the list

File: docs/latest/user-guides/ip-lists/overview.md [hunk 5]
```diff
@@ -8,30 +8,94 @@ In the **IP lists** section of Wallarm Console, you can control access to your a
...
+## Automatic listing
+
+You can enable Wallarm to denylist or graylist IP addresses automatically if they produce some suspicious traffic. This can be done for:
+
+* [Brute force protection](../../admin-en/configuration-guides/protecting-against-bruteforce.md)
+* [BOLA Protection](../../admin-en/configuration-guides/protecting-against-bola.md)
+* Exceeded threshold for malicious payloads
+* [API Abuse Prevention](../../about-wallarm/api-abuse-prevention.md#how-api-abuse-prevention-works)
+
+
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/ip-lists/overview.md:72]: extra new line

File: docs/latest/user-guides/ip-lists/overview.md [hunk 6]
```diff
@@ -8,30 +8,94 @@ In the **IP lists** section of Wallarm Console, you can control access to your a
...
+* For **API Abuse Prevention** - immediately
+
+## Events from denylisted IPs
+
+You can access a list of events originating from any IP address currently on the denylist or previously listed, by clicking on the IP address of interest. This will redirect you to the Events section, where the [relevant event list](../events/analyze-attack.md#analyze-requests-from-denylisted-ips) is displayed.
+
+![Img](../../images/user-guides/events/denylisted-ip-events.png)
+
+## IP lists vs load balancers and CDNs
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/ip-lists/overview.md:87]: this should be in the beginning as it is mandatory for correct operation of IP lists

File: docs/latest/user-guides/ip-lists/overview.md [hunk 7]
```diff
@@ -8,30 +8,94 @@ In the **IP lists** section of Wallarm Console, you can control access to your a
...
+* [Proofpoint Emerging Threats Rules](https://rules.emergingthreats.net/blockrules/compromised-ips.txt)
+* [DigitalSide Threat-Intel Repository](http://osint.digitalside.it/Threat-Intel/lists/latestips.txt)
+* [GreenSnow](https://blocklist.greensnow.co/greensnow.txt)
+* [www.blocklist.de](https://www.blocklist.de/en/export.html)
+* [NGINX ultimate bad bot blocker](https://github.com/mitchellkrogza/nginx-ultimate-bad-bot-blocker/blob/master/_generator_lists/bad-ip-addresses.list)
+* [IPsum](https://github.com/stamparm/ipsum)
+
+## Limiting by target application
+
+When adding object to a list, by default it will process all requests from the listed IP. But you can limit that by target applications: select one or several applications and only requests from the listed IP to that applications will be processed.
```
[the remark this answers is not in the document]
Comment (MaximBashurov) [reply docs/latest/user-guides/ip-lists/overview.md:49]: It does work this way. See UI and in the current docs there is a step: "Select the applications to which you allow or restrict access for the specified IP addresses." for each list.

File: docs/latest/user-guides/ip-lists/overview.md (hunk 1 above)
Comment (MaximBashurov) [reply docs/latest/user-guides/ip-lists/overview.md:51]: I added and marked the "History" term within the section in bold. However, it is not a "mix up" of topics - it is a logical combination. The fact that IP can be not only added but also removed from a list is defined mostly by "period of stay" parameter, not the manual actions. Thus, the main reason for the lists to have "history" is that this parameter is specified.

File: docs/latest/user-guides/ip-lists/overview.md (hunk 2 above)
Comment (MaximBashurov) [reply docs/latest/user-guides/ip-lists/overview.md:11]: I disagree - users need to understand which list to populate and what will happen BEFORE populating.

File: docs/latest/user-guides/ip-lists/overview.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (MaximBashurov) [reply docs/latest/user-guides/ip-lists/overview.md:28]: "Listing" commonly = "adding to list" and listing is better here as we do not speak now about one particular list, but speak about 3 different.

[the remark this answers is not in the document]
Comment (MaximBashurov) [reply docs/latest/user-guides/ip-lists/overview.md:28]: "Automatic listing" deserves, needs, and has a separate section.

File: docs/latest/user-guides/ip-lists/overview.md (hunk 6 above)
Comment (MaximBashurov) [reply docs/latest/user-guides/ip-lists/overview.md:87]: I moved the section to the beginning, but I dislike the result: it really contradicts the comment of "quick starting the population of lists".

File: docs/latest/user-guides/ip-lists/overview.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (AnastasiaTWW) [reply docs/latest/user-guides/ip-lists/overview.md:56]: re-write the first sentence

File: docs/latest/user-guides/ip-lists/overview.md [hunk 8]
```diff
@@ -1,4 +1,4 @@
-# Types and core logic of IP lists
+# Restrictions by IP
 
 In the **IP lists** section of Wallarm Console, you can control access to your applications by allowlisting, denylisting, and graylisting IP addresses.
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/ip-lists/overview.md:4]: in the beginning, in the graylist overview, mention the mode when it is applied

File: docs/latest/user-guides/ip-lists/overview.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (AnastasiaTWW) [reply docs/latest/user-guides/ip-lists/overview.md:35]: at least, mentiuon automatics listing here

File: docs/latest/user-guides/ip-lists/overview.md [hunk 9]
```diff
@@ -8,30 +8,86 @@ In the **IP lists** section of Wallarm Console, you can control access to your a
...
+
+* In any mode, if IP is found in the earlier list, the next one is not considered.
+* Denylisted IPs are blocked even when the node is in `off` or `Monitoring` mode.
+* Graylist is only considered in `Safe blocking` mode.
+
+!!! warning "Exceptions"
+    If [`wallarm_acl_access_phase off`](../../admin-en/configure-parameters-en.md#wallarm_acl_access_phase), the Wallarm node does not analyze the denylist in the `off` mode and does not block requests from denylisted IPs in the `Monitoring` mode.
+
+## Listing IPs, subnets, locations, and source types
+
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/ip-lists/overview.md:36]: + screenshot demonstarting the object listing process

**Pull request closed without merging**